### DISCUSSION TITLE
iOS 端. 版本4.3.33.

### DISCUSSION THREAD

**Discussion by China-jxsg-YanYuanJie** [Tencent/libpag#2105] [Q&A]
播放中的状态退出到后台.等待时间长一些1分钟到2分钟.进入前台. 发现动画停止了. 执行play()也没有反应.打印播放状态是true.但是无论执行暂停重新播放什么各种操作,都无法重新播放

**Comment by China-jxsg-YanYuanJie** [comment]
追述发现的问题
![image](https://github.com/Tencent/libpag/assets/112070751/733580cb-bfcd-4002-a253-f8062f3d30ff)
chang尝试修复的过程发现.
后台过久后(1-2分钟),在即将进入前台的监听方法中去直接调用play()会导致资源异常,重新删除当前pagView 新建一个也不行,必须得在这个方法中延迟一定的时间后,调用play()才可以. ios这边看不到源码.不知道大佬你们内部是做了什么操作无法分析,只能上报这个情况,望给出解决方案

**Reply by petyou** [reply]
遇到同样的问题了

**Reply by petyou** [reply]
最新版本'4.3.47'还是这样，必现的问题

**Reply by kevingpqi123** [reply]
pag 的源码是开源的哈，到源码仓库中可以查看到相关实现。
针对你们遇到的问题，为了提高对接效率，建议你们基于 pag 的官方 demo 提供复现场景及复现步骤，人力有限，我们没有太多的时间帮大家构造场景

**Reply by petyou** [reply]
@kevingpqi123 官方demo即可复现啊，就是demo中的第一个页面，在播放的时候，你退到后台，静止15s，再回到前台，就停止播放了。并且点击无法再次播放
![Uploading 20240124-103449.jpeg…]()

**Reply by petyou** [reply]
很好复现，且必现。或者你播放的时候直接让手机息屏15s也可以复现

**Reply by petyou** [reply]
我们是刚开始接入PAG，现在卡主了。。。

**Reply by China-jxsg-YanYuanJie** [reply]
@petyou 尝试了.监听didBecomeActiveNotification 这个事件里面再让PagView 播放是可以的.需要app进入活跃状态之后改变是可以的.否者会导致pagView 进入异常状态

**Reply by petyou** [reply]
@kevingpqi123 你复现的时候不要连着xcode调试。断开它

**Reply by petyou** [reply]
@China-jxsg-YanYuanJie  这个方式确实可以，如果官方来不及修复。先这样处理了。
但感觉还是SDK内部修复会更好一些

**Reply by kevingpqi123** [reply]
建议你们提供复现 demo，我这边没有时间帮你们构建场景，或者基于 pag 的 官方 demo 做测试，我们对接问题抽的都是自己的空余时间，同时不能影响自己的开发进度，为了提高对接效率，我们只分析提供信息完全的问题

**Reply by petyou** [reply]
@kevingpqi123 大佬，理解你们的意思。
我们上面描述的问题，我基于官方提供的demo，可以100%复现。就是官方demo中的这个页面。

![20240124-103449](https://github.com/Tencent/libpag/assets/18586502/7e6e1f7c-aabe-41af-b6c5-bf81c42bd735)

复现步骤为：打开官方最新的demo，点击 `A simple PAG Animation`，进入动画页面。锁屏手机，静止20s。解锁手机，demo中的动画就停止了。点击页面也不会再播放。

**Reply by petyou** [reply]
demo下载地址：https://pag.art/docs/pad-demo-download.html

**Comment by CodeJhF** [comment]
感谢 @China-jxsg-YanYuanJie 的回复。

因长期无回复，且 libpag 4.3 及更早版本已停止维护（当前仅维护 4.4 / 4.5），先关闭本条。
上报版本：4.3.33。

如果问题在最新版本（4.4 / 4.5）仍能复现，欢迎在本帖回复或新开 issue，并附上 libpag 版本号、平台与系统版本、可复现的 PAG 文件或最小 demo，我们会优先跟进。（stale-close）
